### PR TITLE
Pass directory when launching win32 app; Allow launching win32 apps with admin permissions

### LINKED ISSUE #205: If target game/app needs Admin permission, it won't run
Bluestacks only runs nicely if executed with Administrator rights. I set this straight to the HD-Player.exe executable, so when I launch it from Steam, I get the security prompt to execute it.

On the GlosSI target, though, the security prompt never shows up. Not only that, but GlosSI itself keeps running as if waiting for the executable that never shows up.

## Expected Behavior

When running the game/app that requires Admin rights, I'd expect the security prompt to show up so I can authorize it. After that, the game/app would be launched and picked up by GlosSI.

## Actual Behavior

Nothing happens.

## Steps to Reproduce the Problem

  1. Set a game/app to require being run as Admin by right-clicking it's executable -> Properties -> Compatibility -> check "Run this program as admin" option.
  2. Add it to GlosSI and from GlosSI to Steam.
  3. Try to run the game on Steam.

## General Information

  - Version: (0.1.1.2)
  - Platform: (Win11 Desktop)
  - Controller(s): (1 Steam Controller)
  - Application: (GlosSIConfig)

## Logs / Additional Files

I see nothing useful on the logs, but here they are:

- config, default.json
```json
{
    "controller": {
        "allowDesktopConfig": false,
        "emulateDS4": false,
        "maxControllers": 1
    },
    "devices": {
        "hideDevices": true,
        "realDeviceIds": false
    },
    "extendedLogging": false,
    "icon": null,
    "launch": {
        "closeOnExit": true,
        "ignoreLauncher": true,
        "killLauncher": false,
        "launch": false,
        "launchAppArgs": null,
        "launchPath": null,
        "launcherProcesses": [
        ],
        "waitForChildProcs": true
    },
    "name": null,
    "snapshotNotify": false,
    "steamgridApiKey": "6bd9d6dfa566adfe6a70d7ac9d08ed4f",
    "version": 1,
    "window": {
        "disableOverlay": false,
        "maxFps": null,
        "scale": null,
        "windowMode": false
    }
}
```

- log, GlosSIWatchdog.log
```
[2022-12-29 23:35:29.302] [log] [info] GlosSIWatchdog loaded
[2022-12-29 23:35:29.302] [log] [info] Version: 0.1.1.2
[2022-12-29 23:35:29.303] [log] [debug] Found GlosSITarget window; Starting watch loop
[2022-12-29 23:35:30.322] [log] [debug] Received settings from GlosSITarget: {"controller":{"allowDesktopConfig":false,"emulateDS4":false,"maxControllers":1},"devices":{"hideDevices":true,"realDeviceIds":false},"extendedLogging":false,"icon":"C:/Program Files/BlueStacks_nxt/HD-Player.exe","launch":{"closeOnExit":true,"launch":true,"launchAppArgs":"--instance Nougat64","launchPath":"C:/Program Files/BlueStacks_nxt/HD-Player.exe","waitForChildProcs":true},"name":"HD-Player","version":1,"window":{"disableOverlay":false,"maxFps":0,"scale":0.0,"windowMode":false}}
[2022-12-29 23:35:30.323] [log] [trace] Err parsing "killLauncher"; [json.exception.out_of_range.403] key 'killLauncher' not found
[2022-12-29 23:35:30.323] [log] [trace] Err parsing "ignoreLauncher"; [json.exception.out_of_range.403] key 'ignoreLauncher' not found
[2022-12-29 23:35:36.744] [log] [error] Couldn't fetch launched PIDs: 2
[2022-12-29 23:35:37.078] [log] [info] GlosSITarget was closed. Resetting HidHide state...
[2022-12-29 23:35:37.078] [log] [info] Closing launched processes
[2022-12-29 23:35:37.078] [log] [info] Unloading Watchdog...
```

- log, glossitarget.log
```
[2022-12-29 23:35:28.251] [log] [info] GlosSITarget version: 0.1.1.2
[2022-12-29 23:35:28.259] [log] [debug] Read config file "C:\Users\johnnyc\AppData\Roaming\GlosSI\Targets\HD-Player.json"; config: {"controller":{"allowDesktopConfig":false,"emulateDS4":false,"maxControllers":1},"devices":{"hideDevices":true,"realDeviceIds":false},"extendedLogging":false,"icon":"C:/Program Files/BlueStacks_nxt/HD-Player.exe","launch":{"closeOnExit":true,"ignoreLauncher":true,"killLauncher":false,"launch":true,"launchAppArgs":"--instance Nougat64","launchPath":"C:/Program Files/BlueStacks_nxt/HD-Player.exe","launcherProcesses":[],"waitForChildProcs":true},"name":"HD-Player","snapshotNotify":false,"version":1,"window":{"disableOverlay":false,"maxFps":null,"scale":null,"windowMode":false}}
[2022-12-29 23:35:28.260] [log] [info] Running on Windows 11; Winver: 10.0.22621
[2022-12-29 23:35:28.260] [log] [info] Detected Steam Path: d:/program files (x86)/steam
[2022-12-29 23:35:28.260] [log] [info] Detected Steam UserId: 42758397
[2022-12-29 23:35:28.282] [log] [info] Detected Overlay hotkey(s): Shift+KEY_TAB
[2022-12-29 23:35:28.292] [log] [debug] Connected to ViGEm
[2022-12-29 23:35:28.300] [log] [info] Detected screenshot hotkey(s): KEY_NONE
[2022-12-29 23:35:28.805] [log] [info] Detected resolution: 2560x1080
[2022-12-29 23:35:28.805] [log] [info] Creating Overlay window (Borderless Fullscreen)...
[2022-12-29 23:35:29.259] [log] [debug] Full screen size: 4480x2160
[2022-12-29 23:35:29.259] [log] [debug] Primary monitor size: 2560x1080
[2022-12-29 23:35:29.259] [log] [debug] Window position: 0x0
[2022-12-29 23:35:29.259] [log] [info] Resizing window to 1px smaller than fullscreen...
[2022-12-29 23:35:29.261] [log] [debug] Screen DPI: 96
[2022-12-29 23:35:29.262] [log] [trace] Limiting FPS to 59
[2022-12-29 23:35:29.271] [log] [debug] auto screen sCale: 1
[2022-12-29 23:35:29.272] [log] [debug] Not applying too low screen scale setting
[2022-12-29 23:35:29.272] [log] [trace] Set process priority to HIGH_PRIORITY_CLASS
[2022-12-29 23:35:29.272] [log] [debug] Unpatching Valve CreateProcess Hooks
[2022-12-29 23:35:29.272] [log] [trace] Patching "CreateProcessW"...
[2022-12-29 23:35:29.272] [log] [trace] Unpatched "CreateProcessW"
[2022-12-29 23:35:29.272] [log] [debug] App launch requested
[2022-12-29 23:35:29.282] [log] [debug] Successfully injected
[2022-12-29 23:35:29.282] [log] [info] Successfully injected UWPOverlayEnablerDLL.dll into explorer.exe
[2022-12-29 23:35:29.282] [log] [info] Steam-overlay detected.
[2022-12-29 23:35:29.282] [log] [warning] Double press Steam- overlay key(s)/Controller button to show GlosSI-overlay
[2022-12-29 23:35:29.301] [log] [debug] Successfully injected
[2022-12-29 23:35:29.301] [log] [info] Successfully injected GlosSIWatchdog.dll into explorer.exe
[2022-12-29 23:35:29.319] [log] [debug] Setting up device hiding...
[2022-12-29 23:35:29.319] [log] [debug] Unpatching Valve HID hooks...
[2022-12-29 23:35:29.319] [log] [trace] Patching "SetupDiEnumDeviceInfo"...
[2022-12-29 23:35:29.319] [log] [trace] Unpatched "SetupDiEnumDeviceInfo"
[2022-12-29 23:35:29.319] [log] [trace] Patching "SetupDiGetClassDevsW"...
[2022-12-29 23:35:29.319] [log] [trace] Unpatched "SetupDiGetClassDevsW"
[2022-12-29 23:35:29.319] [log] [trace] Patching "HidD_GetAttributes"...
[2022-12-29 23:35:29.319] [log] [trace] Unpatched "HidD_GetAttributes"
[2022-12-29 23:35:29.319] [log] [trace] Patching "HidD_GetPreparsedData"...
[2022-12-29 23:35:29.319] [log] [trace] Unpatched "HidD_GetPreparsedData"
[2022-12-29 23:35:29.319] [log] [trace] Patching "HidD_GetProductString"...
[2022-12-29 23:35:29.319] [log] [trace] Unpatched "HidD_GetProductString"
[2022-12-29 23:35:29.320] [log] [trace] Patching "HidP_GetButtonCaps"...
[2022-12-29 23:35:29.320] [log] [trace] Unpatched "HidP_GetButtonCaps"
[2022-12-29 23:35:29.320] [log] [trace] Patching "HidP_GetCaps"...
[2022-12-29 23:35:29.320] [log] [trace] Unpatched "HidP_GetCaps"
[2022-12-29 23:35:29.320] [log] [trace] Patching "HidP_GetData"...
[2022-12-29 23:35:29.320] [log] [trace] Unpatched "HidP_GetData"
[2022-12-29 23:35:29.320] [log] [trace] Patching "HidP_GetUsageValue"...
[2022-12-29 23:35:29.320] [log] [trace] Unpatched "HidP_GetUsageValue"
[2022-12-29 23:35:29.320] [log] [trace] Patching "HidP_GetUsages"...
[2022-12-29 23:35:29.320] [log] [trace] Unpatched "HidP_GetUsages"
[2022-12-29 23:35:29.320] [log] [trace] Patching "HidP_GetValueCaps"...
[2022-12-29 23:35:29.320] [log] [trace] Unpatched "HidP_GetValueCaps"
[2022-12-29 23:35:29.350] [log] [info] Hid Gaming Devices; Enabling Overlay element...
[2022-12-29 23:35:29.351] [log] [info] LaunchApp is Win32, launching...
[2022-12-29 23:35:29.351] [log] [debug] Launching Win32App app "C:\Program Files\BlueStacks_nxt\HD-Player.exe"; args ""C:\Program Files\BlueStacks_nxt\HD-Player.exe" --instance Nougat64"
[2022-12-29 23:35:29.368] [log] [error] Couldn't start program: "C:\Program Files\BlueStacks_nxt\HD-Player.exe"
[2022-12-29 23:35:29.368] [log] [debug] Hooking GetForegroundWindow (in own process)
[2022-12-29 23:35:30.245] [log] [info] Configured to close on all children exit. Shutting down...
[2022-12-29 23:35:31.377] [log] [info] Plugged in controller 0, 1; VID: 28de; PID: 28e
[2022-12-29 23:35:33.288] [log] [debug] Started http-server on port 8756
[2022-12-29 23:35:33.288] [log] [info] Un-hid Gaming Devices
```

- log, glossiconfig.log
```
[Thu Dec 29 23:35:51 2022] Warning: qrc:/qml/ShortcutProps.qml:166: ReferenceError: closeOnExit is not defined
[Thu Dec 29 23:35:51 2022] Warning: qrc:/qml/AdvancedTargetSettings.qml:547: TypeError: Type error
[Thu Dec 29 23:35:51 2022] Warning: qrc:/qml/AdvancedTargetSettings.qml:547: TypeError: Type error
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/SteamGridDialog.qml:141: TypeError: Cannot read property 'steamgridOutput' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/AddSelectTypeDialog.qml:104: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/AddSelectTypeDialog.qml:95: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/AddSelectTypeDialog.qml:87: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/main.qml:556: TypeError: Cannot read property 'versionString' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutProps.qml:348: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutProps.qml:328: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutProps.qml:255: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutProps.qml:248: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutProps.qml:72: TypeError: Cannot read property 'getGridImagePath' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/main.qml:389: TypeError: Cannot read property 'targetList' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/main.qml:288: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/main.qml:129: TypeError: Cannot read property 'newVersionName' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/main.qml:137: TypeError: Cannot read property 'newVersionName' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/main.qml:36: TypeError: Cannot read property 'hasAcrlyicEffect' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:78: TypeError: Cannot read property 'isInSteam' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:143: TypeError: Cannot read property 'getAppId' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:136: TypeError: Cannot read property 'isDebug' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:130: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:119: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:79: TypeError: Cannot read property 'getGridImagePath' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:78: TypeError: Cannot read property 'isInSteam' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:143: TypeError: Cannot read property 'getAppId' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:136: TypeError: Cannot read property 'isDebug' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:130: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:119: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:79: TypeError: Cannot read property 'getGridImagePath' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:78: TypeError: Cannot read property 'isInSteam' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:143: TypeError: Cannot read property 'getAppId' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:136: TypeError: Cannot read property 'isDebug' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:130: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:119: TypeError: Cannot read property 'isWindows' of null
[Thu Dec 29 23:35:59 2022] Warning: qrc:/qml/ShortcutCards.qml:79: TypeError: Cannot read property 'getGridImagePath' of null
```

### PR DESCRIPTION
**Pull request by jglamine**
This PR contains two fixes.

1. Pass directory when launching win32 apps.

I have a game which crashes if the working directory isn't the same as the directory the executable is in. This change always passes that directory as the working directory when launching win32 apps.

2. Allow launching win32 apps with admin permissions.

I have a game which requires admin permissions. If I manually add it to steam it works (windows pops up the UAC prompt for me to accept, then launches the game). But when launched through GlosSI, it fails to start up.

It turns out `CreateProcessW` was failing with `ERROR_ELEVATION_REQUIRED` (error code 740).

If `CreateProcessW` fails with that error, I use `ShellExecuteExW` with the `runas` verb to launch the game. This way windows shows the UAC prompt and the game can run with admin permissions.

With this change, games that don't require admin permissions are still launched with `CreateProcessW`. It's only if a game fails to launch that I fall back on `ShellExecuteExW`.

This fixes #205

### CODE REVIEW

File: GlosSITarget/AppLauncher.cpp [hunk 1]
```diff
@@ -24,6 +24,7 @@ limitations under the License.
 #include <Propsys.h>
 #include <propkey.h>
 #include <shellapi.h>
+#include <winerror.h>
```
Comment (jglamine) [inline GlosSITarget/AppLauncher.cpp:27]: This contains the ERROR_ELEVATION_REQUIRED constant

**Comment by Alia5** [comment]
Thank you very much for your contribution!

Fix 2 looks good to me.

---

I have issues with _"Fix" 1_, however.

As you may or may not have noticed, Passing the path to "createProcess" was in GlosSI once, but I took it out to allow user customization in Steams shortcut properties.   
See: https://github.com/Alia5/GlosSI/commit/1a47ef8c351570681516485d63039d6f580fd389  
It's been like that for ~15 months and never caused issue to my knowledge.  
You could have got that from the git-blame btw ;P 

So please take this patch out of this PR.

If you have a fringe case where this does cause issue, though, please open a **separate** PR for this **and** provide a json config-parameter for this.

---

Additionally, please rebase this PR onto develop

**Comment by jglamine** [comment]
Thanks for such a quick review! Oh cool, I didn't realize that if I set "Start In" in the steam shortcut that gets used.

I removed the launch directory stuff.

**Comment by InanimateCrbnRod** [comment]
I'm confused on how to apply this fix

**Comment by Alia5** [comment]
Download the latest snapshot build...

**Pull request merged**